types.d.ts

```diff
@@ -0,0 +1,59 @@
+// this file was automatically generated, DO NOT EDIT
```

Comment:
I don't like the location of this file. Could we put it with package under `npm` or `js` or a similar directory?

Reply:
I initially put it in `docs` because I forgot the second part of this comment. I just moved it to a new `types` directory.